src/figuresClasses.ts

```diff
@@ -1,11 +1,71 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
```

Comment:
[CODE KNOWLEDGE] - You don't need to manually assign the constructor parameters to class properties when using the access modifiers 'public' or 'private'. TypeScript does this automatically. You can remove the assignments and leave the constructor body empty if there are no other operations to perform.